src/Objs/Redirect/RedirectComponent.tsx

```diff
@@ -21,6 +21,8 @@ provideComponent(Redirect, ({ page }) => {
     })
   }, [page])
 
+  if (page.get('ensureUserIsLoggedIn')) ensureUserIsLoggedIn()
```

Comment:
* [ ] From the method name I'd assume this call has a side effect, so we should put it in an effect. 
* [ ] We must make sure not to create a race between `ensureUserIsLoggedIn()` and `location.replace`, because a reload will undo whatever `ensureUserIsLoggedIn` triggers. I think a fix would include branching depending on `isUserLoggedIn()`.